app/models/appointment.rb

```diff
@@ -1,5 +1,34 @@
 class Appointment < ActiveRecord::Base
 	belongs_to :doctor
 	belongs_to :patient
-	validates_presence_of :doctor
+
+  def bookable?
+    doctor_id = self.doctor_id
+    schedule = self.schedule
+    appointment_count = Appointment.where(doctor_id: doctor_id, status:'Approved', schedule: schedule ).count
+    if appointment_count < 4
+      return false
```

Comment:
this is implied by default. `appointment_count < 4` itself is sufficient. It is a condition, and returns boolean.